src/MOI_wrapper/MOI_wrapper.jl

```diff
@@ -126,367 +171,236 @@ function MOI.is_empty(model::Optimizer)
     return Cbc_getNumCols(model) == 0 && Cbc_getNumRows(model) == 0
 end
 
-mutable struct _CbcModelFormat
-    num_rows::Cint
-    num_cols::Cint
-    # (row_idx, col_idx, values) are the sparse elements of the constraint matrix.
-    row_idx::Vector{Cint}
-    col_idx::Vector{Cint}
-    values::Vector{Float64}
-    # Constraint bounds.
-    row_lb::Vector{Float64}
-    row_ub::Vector{Float64}
-    # Variable bounds.
-    col_lb::Vector{Float64}
-    col_ub::Vector{Float64}
-    # Columns that are binary or integer
-    binary::Vector{Int}
-    integer::Vector{Int}
-    # SOS1 constraints
-    sos1_starts::Vector{Cint}
-    sos1_indices::Vector{Cint}
-    sos1_weights::Vector{Float64}
-    # SOS2 constraints
-    sos2_starts::Vector{Cint}
-    sos2_indices::Vector{Cint}
-    sos2_weights::Vector{Float64}
-    # Objective coefficients.
-    obj::Vector{Float64}
-    objective_constant::Float64
-
-    # An `InexactError` might occur if `num_rows` or `num_cols` is too
-    # large, e.g., if `num_cols isa Int64` and is larger than 2^31 on
-    # 32-bit hardware.
-    function _CbcModelFormat(num_rows::Cint, num_cols::Cint)
-        return new(
-            num_rows,
-            num_cols,
-            # Constraint matrix.
-            Cint[],
-            Cint[],
-            Float64[],
-            # Row lower/upper bounds.
-            fill(-Inf, num_rows),
-            fill(Inf, num_rows),
-            # Column lower/upper bounds.
-            fill(-Inf, num_cols),
-            fill(Inf, num_cols),
-            # Binary/Integer columns.
-            Cint[],
-            Cint[],
-            # SOSI constraints.
-            Cint[],
-            Cint[],
-            Float64[],
-            # SOSII constraints.
-            Cint[],
-            Cint[],
-            Float64[],
-            # Objective vector and offset.
-            fill(0.0, num_cols),
-            0.0,
-        )
-    end
+function _index_map(
+    src::OptimizerCache,
+    ::MOI.IndexMap,
+    ci::MOI.ConstraintIndex{<:MOI.ScalarAffineFunction},
+)
+    return MOI.Utilities.rows(src.constraints.moi_equalto, ci)
```

Comment:
Isn't that `ci.value`?

Reply:
No 😢 that's why we need a few index maps.

Reply:
Ah yes, it would be if you were using `MixOfScalarSets` instead. If you use `OrderedProductOfSets` and want an identity map then you need to do the mapping by keeping the product of sets after the copy, see e.g.,
https://github.com/jump-dev/ECOS.jl/blob/e246e6f3de8ce7b54daebef1090b4c5dc16e0cfc/src/MOI_wrapper/MOI_wrapper.jl#L91-L92

Reply:
I think this is fine as-is.

Reply:
Yes, we can keep it as is. But it's good to keep in mind for what we should recommend in the MOI docs.